sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGenerateExec.scala

```diff
@@ -528,13 +532,35 @@ abstract class GpuExplodeBase extends GpuUnevaluableUnaryExpression with GpuGene
   /**
    * A function that will do the explode or position explode
    */
-  private[this] def explodeFun(inputTable: Table, genOffset: Int, outer: Boolean): Table = {
+  private[this] def explodeFun(
+      inputTable: Table,
+      genOffset: Int,
+      outer: Boolean,
+      fixUpOffset: Long): Table = {
     if (position) {
-      if (outer) {
+      val toFixUp = if (outer) {
         inputTable.explodeOuterPosition(genOffset)
       } else {
         inputTable.explodePosition(genOffset)
       }
+      // the position column is at genOffset and the exploded column at genOffset + 1
+      withResource(toFixUp) { _ =>
+        val posColToFix = toFixUp.getColumn(genOffset)
+        val fixedUpPosCol = withResource(Scalar.fromLong(fixUpOffset)) { offset =>
+          posColToFix.add(offset, posColToFix.getType)
+        }
+        withResource(fixedUpPosCol) { _ =>
+          val newCols = new Array[ColumnVector](toFixUp.getNumberOfColumns)
+          (0 until genOffset).foreach { b =>
+            newCols(b) = toFixUp.getColumn(b)
+          }
+          newCols(genOffset) = fixedUpPosCol
+          (genOffset + 1 until toFixUp.getNumberOfColumns).foreach { a =>
+            newCols(a) = toFixUp.getColumn(a)
+          }
```

Comment:
```suggestion
          val newCols = (0 until toFixUp.getNumberOfColumns).map { i =>
            if (i == genOffset) {
              fixedUpPosCol
            } else {
              toFixUp.getColumn(i)
          }.toArray
```

Reply:
Let me make that change, that's better.

Reply:
Thanks @jlowe this is handled in https://github.com/NVIDIA/spark-rapids/pull/10131/commits/123b5d87708040c3bc296e6b5372f617ec66a7d4